src/main/java/edu/kis/powp/jobs2d/command/CompoundCommand.java

```diff
@@ -2,45 +2,69 @@
 
 import edu.kis.powp.jobs2d.Job2dDriver;
 
+import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
 
-import static java.util.stream.Collectors.toList;
 
 public class CompoundCommand implements ICompoundCommand {
 
-    private final String name;
-    private final List<DriverCommand> driverCommands;
+    private String name;
+    private List<DriverCommand> commandsList;
+    private List<DriverCommand> driverCommands;
 
-    public CompoundCommand(List<DriverCommand> commandList, String name) {
+    public CompoundCommand(List<DriverCommand> commands, String name) {
+        super();
+        this.commandsList = new ArrayList<>();
+        commands.iterator().forEachRemaining(command -> this.commandsList.add((DriverCommand) command.clone()));
         this.name = name;
-        this.driverCommands = commandList;
     }
 
+
+    public CompoundCommand(String other, String name) {
+        super();
+        this.name = name;
+        this.commandsList = new ArrayList<>();
+
+    }
+
+
     public List<DriverCommand> getDriverCommands() {
         return driverCommands;
     }
 
     @Override
     public void execute(Job2dDriver driver) {
-        driverCommands.forEach((c) -> c.execute(driver));
+
+        this.iterator().forEachRemaining(command -> command.execute(driver));
     }
 
-    @Override
     public Iterator<DriverCommand> iterator() {
-        return driverCommands.iterator();
+
+        return commandsList.iterator();
     }
 
     @Override
-    public String toString() {
-        return name;
+    public CompoundCommand clone() {
+        CompoundCommand command = null;
+        try {
+            command = (CompoundCommand) super.clone();
+            command.name = this.name;
+            command.commandsList = new ArrayList<>();
+            for (DriverCommand cmd : this.commandsList) {
+                command.commandsList.add((DriverCommand) cmd.clone());
+            }
+        } catch (CloneNotSupportedException e) {
+            command = new CompoundCommand((List<DriverCommand>) this, this.name);
+        }
+        return command;
     }
 
     @Override
-    public Object clone() {
-        return new CompoundCommand(driverCommands.stream()
-                .map(x -> (DriverCommand) x.clone())
-                .collect(toList()), this.name);
+    public String toString() {
+        return name;
```

Comment:
Po co Panowie wprowadzają zmiany w metody, które nie należą do waszej pracy?! Jak ja mam sprawdzać pracę? Czy mają te zmiany znaczenie czy nie? Jeżeli nie, to ich nie powinno tu być. Rozwiązanie powinno być osadzone w najnowszej wersji projektu. Nie rozumiem, czy to efekt zamierzony, czy jakiś błąd w pracy z repo?

Reply:
Oczywiście ma Pan rację co do tego. Próbując wykonać testy wyskakiwały nam błędy cały czas przy naszym wcześniejszym zadaniu i próbowaliśmy to naprawić w całym repozytorium lecz to nie przyniosło oczekiwanych skutków. Jeśli będzie taka potrzeba wrócimy do tych zmian wcześniejszych w tych klasach aby nie powodować niepotrzebnych problemów z kodem.